### DISCUSSION TITLE
interesting solution to my issue

### DISCUSSION THREAD

**Discussion by edwardsmarkf** [katspaugh/wavesurfer.js#2862] [General]
my issue yesterday:   https://github.com/katspaugh/wavesurfer.js/discussions/2849

i believe my solution has a **VERY** interesting application to my favorite audio player which is **waveSurfer** of course.

if somebody wants to experiment with wavesurfer & see it working locally, all you need is the chrome browser.  i have not tried this on FF.

_step one:_   open up the webpage  "**about:blank**" (not sure why this is now necessary, but its the only way i could get it to work).  
you should see a comprehensive list of other web players to consider.  < / humor >

_step two:_  right-click inspect anywhere on the blank page, or just do ctrl-shift-i

_step three:_  enter the following code in the console:

```
var WaveSurfer = null;
let script		= document.createElement('script')		;
script.type		= 'module'				;
script.innerText	= "import ws from 'https://unpkg.com/wavesurfer.js@beta'; WaveSurfer=ws;";
document.head.appendChild(script);
```
_step four_:  wait a half a second and then enter this code
```
const wavesurfer = WaveSurfer.create({
  'container': document.body,
  waveColor: '#4F4A85',
  progressColor: '#383351',
  url: 'https://wavesurfer-js.org//wavesurfer-code/examples/audio/audio.wav',
})
wavesurfer.once('interaction', () => {
  wavesurfer.play()
})
```
the basic wavesurver example should now be visible in your browser.

i would like to add a very special and sincere THANK-YOU to **katspaugh** for providing us all with such an excellent audio player as well as the support over many years now.  few of us in life can truly say that we made a real difference in life, and katspaugh can certainly & boldly make that claim.

**Comment by katspaugh** [comment]
Thanks for sharing! :)

**Comment by edwardsmarkf** [comment]
**Katspaugh** might want to be careful when thanking his minions, otherwise, they might expand the sample to include the timeline:


```
var WaveSurfer = null;
var TimelinePlugin = null;
let innerText = `
    import ws from 'https://unpkg.com/wavesurfer.js@beta'; 
    WaveSurfer = ws;
    import tlPlugin from 'https://unpkg.com/wavesurfer.js@beta/dist/plugins/timeline.js';
    TimelinePlugin = tlPlugin;
`;

let script		= document.createElement('script')		;
script.type		= 'module'				;
Object.assign(script,  { innerText } ); // fancy way of assigning this object a value!
document.head.appendChild(script);
```
// pause here......
```
const wavesurfer = WaveSurfer.create({
  'container': document.body,
  waveColor: '#4F4A85',
  progressColor: '#383351',
  url: 'https://wavesurfer-js.org//wavesurfer-code/examples/audio/audio.wav',
})
wavesurfer.registerPlugin(TimelinePlugin.create());

wavesurfer.once('interaction', () => {
  wavesurfer.play()
})
```

**Comment by katspaugh** [comment]
You mean when thanking wavesurfer OGs! :D
